presto-docs/src/main/sphinx/optimizer/push-down.rst

```diff
@@ -0,0 +1,140 @@
+            Layout: [regionkey_0:bigint, _presto_generated_1:bigint]
+            Estimates: {rows: ? (?), cpu: ?, memory: 0B, network: 0B}
+            _presto_generated_1 := count(*):_presto_generated_1:bigint:bigint
+            regionkey_0 := regionkey:bigint:int8
+
+A number of factors can prevent a push down:
+
+* adding a condition to the query
+* using a different aggregate function without push down support
+* using a connector without the specific function push down support
+
+As a result the explain plan shows the ``Aggregate`` operation being performed
+by Presto. This is a clear sign that now push down to the database is not
+performed, and instead Presto performs the aggregate processing.
+
+.. code-block:: none
+
+ Fragment 0 [SINGLE]
+     Output layout: [regionkey, count]
+     Output partitioning: SINGLE []
+     Stage Execution Strategy: UNGROUPED_EXECUTION
+     Output[regionkey, _col1]
+     │   Layout: [regionkey:bigint, count:bigint]
+     │   Estimates: {rows: ? (?), cpu: ?, memory: ?, network: ?}
+     │   _col1 := count
+     └─ RemoteSource[1]
+            Layout: [regionkey:bigint, count:bigint]
+
+ Fragment 1 [HASH]
+     Output layout: [regionkey, count]
+     Output partitioning: SINGLE []
+     Stage Execution Strategy: UNGROUPED_EXECUTION
+     Aggregate(FINAL)[regionkey]
+     │   Layout: [regionkey:bigint, count:bigint]
+     │   Estimates: {rows: ? (?), cpu: ?, memory: ?, network: ?}
+     │   count := count("count_0")
+     └─ LocalExchange[HASH][$hashvalue] ("regionkey")
+        │   Layout: [regionkey:bigint, count_0:bigint, $hashvalue:bigint]
+        │   Estimates: {rows: ? (?), cpu: ?, memory: ?, network: ?}
+        └─ RemoteSource[2]
+               Layout: [regionkey:bigint, count_0:bigint, $hashvalue_1:bigint]
+
+ Fragment 2 [SOURCE]
+     Output layout: [regionkey, count_0, $hashvalue_2]
+     Output partitioning: HASH [regionkey][$hashvalue_2]
+     Stage Execution Strategy: UNGROUPED_EXECUTION
+     Project[]
+     │   Layout: [regionkey:bigint, count_0:bigint, $hashvalue_2:bigint]
+     │   Estimates: {rows: ? (?), cpu: ?, memory: ?, network: ?}
+     │   $hashvalue_2 := combine_hash(bigint '0', COALESCE("$operator$hash_code"("regionkey"), 0))
+     └─ Aggregate(PARTIAL)[regionkey]
+        │   Layout: [regionkey:bigint, count_0:bigint]
+        │   count_0 := count(*)
+        └─ TableScan[tpch:nation:sf0.01, grouped = false]
+               Layout: [regionkey:bigint]
+               Estimates: {rows: 25 (225B), cpu: 225, memory: 0B, network: 0B}
+               regionkey := tpch:regionkey
+
+Limitations
+-----------
+
+Push down does not support a number of more complex statements:
+
+* complex grouping sets such as ``GROUP BY GOUPING SETS``, ``ROLLUP`` or
+    ``CUBE``
+* expressions, like ``sum(a * b)``
+* coercions, like ``sum(integer_column)``
```

Comment:
```
* coercions: ``sum(integer_column)``
```

This one is non-obvious and is arguably a limitation that we need to fix.

Reply:
sure ... but until its fixed it should be documented like this.